crates/nu-path/src/pwd_per_drive.rs

```diff
@@ -0,0 +1,366 @@
+use std::path::{Path, PathBuf};
+
+cfg_if::cfg_if! { if #[cfg(windows)] {
+
+
+struct Drive2PWDmap {
+    map: [Option<String>; 26], // Fixed-size array for A-Z
+}
+
+impl Drive2PWDmap {
+    pub fn new() -> Self {
+        Drive2PWDmap {
+            map: Default::default(), // Initialize all to `None`
+        }
+    }
+
+    /// Set the PWD for the drive letter in the absolute path.
+    /// Return String for error description.
+    pub fn set_pwd(&mut self, path: &Path) -> Result<(), String> {
+        if let (Some(drive_letter), Some(path_str)) = (
+            Self::extract_drive_letter(path),
+            path.to_str(),
+        ) {
+            self.map[drive_letter as usize - 'A' as usize] = Some(path_str.to_string());
+            return Ok(());
+        }
+        Err(format!("Invalid path: {}", path.display()))
```

Comment:
Please just use an `else` instead of early `return`

Reply:
OK